### PR TITLE
ADEN-4657 | Porvata/vUAP: viewportObserver throws an error on Mercury

### PR DESCRIPTION
**Pull request by mleczey**
## Links
https://wikia-inc.atlassian.net/browse/ADEN-4657
## Description
Porvata/vUAP: viewportObserver throws an error on Mercury

### CODE REVIEW

File: resources/wikia/modules/domCalculator.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ define('wikia.domCalculator', [
 			element = element.offsetParent;
 		} while (element !== null);
 
-		if (elementWindow.frameElement) {
+		if (elementWindow && elementWindow.frameElement) {
```
Comment (dianafa) [inline resources/wikia/modules/domCalculator.js:17]: do we know why `elementWindow` is not present on mercury?

Comment (mleczey) [reply resources/wikia/modules/domCalculator.js:17]: will be described in the ticket

Comment (bognix) [reply resources/wikia/modules/domCalculator.js:17]: how about removing `element.ownerDocument.parentWindow`? It's not supported by any browser we support.

Comment (mleczey) [reply resources/wikia/modules/domCalculator.js:17]: done

File: extensions/wikia/AdEngine/js/video/player/porvata/porvata.js [hunk 2]
```diff
@@ -96,6 +97,15 @@ define('ext.wikia.adEngine.video.player.porvata', [
 					params.onReady(video);
 				}
 
+				if (mercuryListener) {
```
Comment (bognix) [inline extensions/wikia/AdEngine/js/video/player/porvata/porvata.js]: I'm not sure if Porvata should be aware of if it's running on mercury or not. How about doing it other way around, for instance trigger event `allAdsCompleted` from some `mercury-related` file on page change? Or create custom event like `onPlayerDestroy`and dispatch this event on player from some `mercury-related` file

/cc @fraszczakszymon

File: extensions/wikia/AdEngine/js/video/player/porvata/porvataPlayerFactory.js [hunk 3]
```diff
@@ -31,6 +31,7 @@ define('ext.wikia.adEngine.video.player.porvata.porvataPlayerFactory', [
 		return {
 			container: prepareVideoAdContainer(videoAdContainer, params),
 			ima: ima,
+			destroyCallbacks: [],
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/video/player/porvata/porvataPlayerFactory.js]: Can we declare it within scope of `create` method? So it's not accessible on video object

**Comment by fraszczakszymon** [comment]
Do we want to implement destroy logic in ad-engine as well?

**Comment by mleczey** [comment]
Discussed about moving logic to ad-engine. Currently we decided that it will remain as it is.

**Comment by fraszczakszymon** [comment]
LGTM 👍

**Pull request merged**